test/client/dio_interceptor_test.dart

```diff
@@ -0,0 +1,197 @@
+
+    late Dio dio;
+    late DateTime now;
+
+    setUp(() {
+      dio = Dio();
+      now = DateTime.now().toUtc();
+    });
+
+    tearDown(() {
+      dio.close();
+    });
+
+    test('Interceptor adds signature header to GET request', () async {
+      // Arrange
+      final options = RequestsSignatureOptions(
+        clientId: testClientId,
+        clientSecret: testClientSecret,
+        headerName: headerName,
+        signaturePattern: signaturePattern,
+      );
+
+      final interceptor = RequestsSignatureInterceptor(options, dio);
+
+      dio.interceptors.add(interceptor);
+
+      // Act
+      final response = await dio.get(baseEndpoint);
+
+      // Assert
+      expect(response.statusCode, 200);
+
+      expect(response.requestOptions.headers, contains(headerName));
+    });
+
+    test('Interceptor returns OK with auto-retry tuned to false', () async {
+      // Arrange
+      final options = RequestsSignatureOptions(
+        clientId: testClientId,
+        clientSecret: testClientSecret,
+        headerName: headerName,
+        signaturePattern: signaturePattern,
+        disableAutoRetryOnClockSkew: false,
+        clockSkew: Duration(milliseconds: clockskewMS),
+      );
+
+      final dioAdapter = DioAdapter(dio: dio);
+      dio.httpClientAdapter = dioAdapter;
+
+      // Create the interceptor
+      final interceptor = RequestsSignatureInterceptor(
+        options,
+        dio,
+        getTime: (request) {
+          return now.millisecondsSinceEpoch ~/ 1000;
+        },
+      );
+
+      // Add the interceptor to Dio
+      dio.interceptors.add(interceptor);
+
+      // Register the mock response
+      dioAdapter.onGet(baseEndpoint, (request) {
+        options.disableAutoRetryOnClockSkew
+            ? request.reply(HttpStatus.unauthorized, emptyDataInjection)
+            : request.reply(HttpStatus.ok, emptyDataInjection);
+      });
+
+      // Act
+      final response = await dio.get(baseEndpoint);
+
+      // Assert
+      expect(response.statusCode, HttpStatus.ok);
+    });
+
+    test('Auto-retry disabled when clock skew option is disabled', () async {
+      final options = RequestsSignatureOptions(
+        clientId: testClientId,
+        clientSecret: testClientSecret,
+        headerName: headerName,
+        signaturePattern: signaturePattern,
+        disableAutoRetryOnClockSkew: true,
+      );
+
+      final dioAdapter = DioAdapter(dio: dio);
+      dio.httpClientAdapter = dioAdapter;
+
+      // Create the interceptor
+      final interceptor =
+          RequestsSignatureInterceptor(options, dio, getTime: (request) {
+        return now.millisecondsSinceEpoch ~/ 1000;
+      });
+
+      // Add the interceptor to Dio
+      dio.interceptors.add(interceptor);
+
+      int requestCount = 0;
+
+      dioAdapter.onGet(baseEndpoint, (request) {
+        requestCount++;
+        request.reply(HttpStatus.ok, emptyDataInjection);
```

Comment:
Same thing here, it doesn't make sense for you to return an OK if you want to test the auto-retry mechanisms. For the interceptor to consider auto-retrying, you need an unauthorized in the first place.